samtranslator/model/eventsources/push.py

```diff
@@ -743,6 +743,24 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def]
         if explicit_api.get("__MANAGE_SWAGGER") or explicit_api.get("MergeDefinitions"):
             self._add_swagger_integration(explicit_api, api_id, function, intrinsics_resolver)  # type: ignore[no-untyped-call]
 
+        swagger_body = explicit_api.get("DefinitionBody")
+
+        if swagger_body and self.Auth and self.Auth.get("OverrideApiAuth"):
+            # TODO: refactor to remove this cast
+            stage = cast(str, self.Stage)
+            editor = SwaggerEditor(swagger_body)
+            self.add_auth_to_swagger(
+                self.Auth,
+                explicit_api,
+                api_id,
+                self.relative_id,
+                self.Method,
+                self.Path,
+                stage,
+                editor,
+                intrinsics_resolver,
+            )
+            explicit_api["DefinitionBody"] = editor.swagger
```

Comment:
Can you add some comments here? Bit confused about what its actually doing

Reply:
For sure, i'll add some comments in the next commit. Essentially it's just creating the resources needed to call `self.add_auth_to_swagger()` and checking if the `OverrideApiAuth` property is set. 

`add_auth_to_swagger()` is needed because it will call `add_auth_to_method()` in `swagger.py` which checks event source `Auth` and updates the `DefinitionBody` with any security overrides.

then the line `explicit_api["DefinitionBody"] = editor.swagger` saves that change to `DefinitionBody` so it can be detected when transforming the `Api` resource.